crypto/evp_extra/pq_sig_kat_tests_dilithium3.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <openssl/evp.h>
+#include <vector>
+#include "../fipsmodule/evp/internal.h"
+#include "../dilithium/sig_dilithium.h"
+#include "../test/file_test.h"
+#include "../test/test_util.h"
+#include "../rand_extra/pq_custom_randombytes.h"
+#include "./internal.h"
+
+ static void RunTest(FileTest *t)
+{
+  std::string count, mlen, smlen;
+  std::vector<uint8_t> seed, msg, pk, sk, sm;
+
+  ASSERT_TRUE(t->GetAttribute(&count, "count"));
+  ASSERT_TRUE(t->GetBytes(&seed, "seed"));
+  ASSERT_TRUE(t->GetAttribute(&mlen, "mlen"));
+  ASSERT_TRUE(t->GetBytes(&msg, "msg"));
+  ASSERT_TRUE(t->GetBytes(&pk, "pk"));
+  ASSERT_TRUE(t->GetBytes(&sk, "sk"));
+  ASSERT_TRUE(t->GetAttribute(&smlen, "smlen"));
+  ASSERT_TRUE(t->GetBytes(&sm, "sm"));
+
+  //the KAT files generated by the dilithium team use the optional APIs that
```

Comment:
@[dkostic](https://github.com/dkostic) dkostic [27 days ago](https://github.com/awslabs/aws-lc/pull/637/files#r1002063413)
what does it mean optional API?

Reply:
The Dilithium authors added extra APIs than what the NIST call for proposals required.

There are two pairs of sign/verify APIs provided by Dilithium, `pqcrystals_dilithium3_ref_signature` and `pqcrystals_dilithium3_ref_verify` which sign and verify (resp.) a Dilithium3 signature generated for some message m, and `pqcrystals_dilithium3_ref` and `pqcrystals_dilithium3_ref_open`, which sign and verify (resp.) a Dilithium3 signed message (which is the composition of a Dilithium3 and the message m that was signed) for some message m. We have chosen to select the two APIs `pqcrystals_dilithium3_ref_signature` and `pqcrystals_dilithium3_ref_verify` to be consistent with existing EVP signature functionality. This is also consistent with other implementations, such as within [LibOQS](https://github.com/open-quantum-safe/liboqs/blob/main/src/sig/dilithium/sig_dilithium_3.c#L33-L34).